packages/storage-plus/src/keys.rs

```diff
@@ -362,15 +421,67 @@ mod test {
         let k: U64Key = 134u64.into();
         let path = k.key();
         assert_eq!(1, path.len());
-        assert_eq!(134u64.to_be_bytes().to_vec(), path[0].to_vec());
+        assert_eq!(134u64.to_be_bytes(), path[0].as_ref());
     }
 
     #[test]
     fn u32key_works() {
         let k: U32Key = 4242u32.into();
         let path = k.key();
         assert_eq!(1, path.len());
-        assert_eq!(4242u32.to_be_bytes().to_vec(), path[0].to_vec());
+        assert_eq!(4242u32.to_be_bytes(), path[0].as_ref());
+    }
+
+    #[test]
+    fn naked_8key_works() {
+        let k: u8 = 42u8;
+        let path = k.key();
+        assert_eq!(1, path.len());
+        assert_eq!(42u8.to_be_bytes(), path[0].as_ref());
+
+        let k: i8 = 42i8;
+        let path = k.key();
+        assert_eq!(1, path.len());
+        assert_eq!(42i8.to_be_bytes(), path[0].as_ref());
+    }
+
+    #[test]
+    fn naked_16key_works() {
+        let k: u16 = 4242u16;
+        let path = k.key();
+        assert_eq!(1, path.len());
+        assert_eq!(4242u16.to_be_bytes(), path[0].as_ref());
+
+        let k: i16 = 4242i16;
+        let path = k.key();
+        assert_eq!(1, path.len());
+        assert_eq!(4242i16.to_be_bytes(), path[0].as_ref());
+    }
+
+    #[test]
+    fn naked_32key_works() {
+        let k: u32 = 4242u32;
+        let path = k.key();
+        assert_eq!(1, path.len());
+        assert_eq!(4242u32.to_be_bytes(), path[0].as_ref());
+
+        let k: i32 = 4242i32;
+        let path = k.key();
+        assert_eq!(1, path.len());
+        assert_eq!(4242i32.to_be_bytes(), path[0].as_ref());
+    }
+
+    #[test]
+    fn naked_64key_works() {
```

Comment:
Good to cover all these cases...

I know if is just cut-and-paste, but maybe cover all the prefix cases as well?
(Maybe overkill as I can visually inspect they are the same, but test cases ensures no one else breaks this in 8 months in some refactor)

Reply:
Added.